powerdns/repo.sls

```diff
@@ -4,13 +4,25 @@ powerdns_server_repo:
   pkgrepo.managed:
     - humanname: PowerDNS
     {% if salt['grains.get']('os_family') == 'Debian' %}
+    {% if (salt['grains.get']('osfullname') == "Ubuntu" and salt['grains.get']('osmajorrelease') >= 22)
+    or (salt['grains.get']('osfullname') == "Debian" and salt['grains.get']('osmajorrelease') >= 12) %}
```

Comment:
I think this logic would be better fitted in map data https://github.com/saltstack-formulas/powerdns-formula/blob/master/powerdns/osfingermap.yaml#L20

Reply:
The downside is that we'll have to keep that updated for every single release.

Reply:
I think the idea is to have the ability to overwrite based on the supported platforms.

you can also keep jinja yaml maps as well https://github.com/saltstack-formulas/salt-formula/blob/master/salt/osmap.yaml#L5-L16